src/main/java/nl/andrewl/emaildatasetbrowser/view/search/export/LuceneSearchExportPanel.java

```diff
@@ -0,0 +1,90 @@
+package nl.andrewl.emaildatasetbrowser.view.search.export;
+
+import java.awt.*;
+import java.util.HashMap;
+
+import javax.swing.*;
+
+import nl.andrewl.emaildatasetbrowser.view.ProgressDialog;
+import nl.andrewl.emaildatasetbrowser.view.search.LuceneSearchPanel;
+
+public class LuceneSearchExportPanel extends JDialog {
+    private static final String PANEL_TITLE = "Lucene Search Export";
+
+    private final JPanel contentPanel = new JPanel();
+
+    private final HashMap<String, QueryExportParameterPanel> parameterPanels = new HashMap<>();
+    private String currentExportPanel;
+
+    public LuceneSearchExportPanel(Window owner, LuceneSearchPanel searchPanel) {
+        super(owner, PANEL_TITLE, ModalityType.TOOLKIT_MODAL);
```

Comment:
I think you want to use `ModalityType.APPLICATION_MODAL`, since apparently there's a whole lot of extra stuff going on with the toolkit modal which we don't need:
> A TOOLKIT_MODAL dialog blocks all top-level windows run from the same toolkit except those from its own child hierarchy. If there are several applets launched in a browser, all of them run with the same toolkit; thus, a toolkit-modal dialog displayed by an applet may affect other applets and all windows of the browser instance which embeds the Java runtime environment for this toolkit. Special AWTPermission "toolkitModality" must be granted to use toolkit-modal dialogs. If a TOOLKIT_MODAL dialog is being created and this permission is not granted, a SecurityException will be thrown, and no dialog will be created. If a modality type is being changed to TOOLKIT_MODAL and this permission is not granted, a SecurityException will be thrown, and the modality type will be left unchanged.

Whereas the APPLICATION_MODAL:
> An APPLICATION_MODAL dialog blocks all top-level windows from the same Java application except those from its own child hierarchy. If there are several applets launched in a browser, they can be treated either as separate applications or a single one. This behavior is implementation-dependent.